### ISSUE TITLE
Migrate the code instead

### ISSUE THREAD

**Issue by TomasVotruba** [laminas/laminas-zendframework-bridge#18]
I see the implementation basically replaces "Zend" with "Laminas" in namespace. This step introduces possible conflict of 2 classes, one with Laminas, one with Zend. It also adds extra work with eventual migration.

Why not migrate the namespace when laminas packages are used already?

I work on a tool that instantly migrates PHP code - [Rector](https://github.com/rectorphp/rector). It can do this migration as well:

```yaml
# configure rector.yaml
services:
    Rector\Rector\Namespace_\RenameNamespaceRector:
        Zend: Laminas
```

```bash
# run
vendor/bin/rector process src/SomeOldZendCode
```

## Result

```diff
-use Zend\Something;
+use Laminas\Something;

 class SomeController
 {
     public function run()
     {
-        return new Zend\Response;
+        return new Laminas\Response;
     }
 }
```

**Comment by JanMikes** [comment]
I know rector and use it regularly on my projects, it is a very good tool for refactoring! 

I have read somewhere that there will be a migration tool from Zend to Laminas, I am curious, what tool would you use, or write a new one?

Imho using something that already exists might save you a lot of time 😉

**Comment by weierophinney** [comment]
@TomasVotruba and @JanMikes —

I'm sorry I didn't see and respond to this earlier. When we pushed the package, we were in the early stages of development.

There is more going on than what you describe, however.

The Laminas packages will act as _replacements_  for the ZF packages. We do not expect there to be a mix of Laminas and ZF packages in a project. However, if there are, we need to do a few things:

- If a reference is made to a ZF class (such as typehinting on an interface), but the replacement Laminas package is installed, _inheritance must still work_. As such, the autoloading in this package also creates `class_alias` statements from the ZF artifacts to the Laminas implementations when they are found, to ensure that inheritance and hinting continues to work.

- Additionally, with dependency configuration, `::class` annotation _does not trigger autoloading_. As such, this package now also provides configuration post-processing in order to rewrite such references to their Laminas equivalents. (It also identifies known ZF configuration keys and rewrites those.)

And that's just what _this_ tool does. The tool for rewriting repositories does far more, as does the tool for migrating userland code to the new packages.

Rector might have helped, but there's a lot of small edge cases that we were able to identify and handle that we may not have been able to shoe-horn into another tool. That said, if I'd seen this earlier, we might have investigated Rector more closely earlier on.
